src/library_manager/lib_manager.c

```diff
@@ -486,15 +486,27 @@ static void lib_manager_update_sof_ctx(struct sof_man_fw_desc *desc, uint32_t li
 }
 
 #if CONFIG_INTEL_MODULES
-int lib_manager_register_module(struct sof_man_fw_desc *desc, int module_id)
+int lib_manager_register_module(const uint32_t component_id)
```

Comment:
A bit more background on motivation of this change would be good. Just looking at this patch, the ipc4_get_comp_drv() is still calculating the module descriptor pointer, and now lib_manager_register_module() does the same again. I feel I'm missing some context.

Reply:
@kv2019i In an upcoming PR, I will simplify the function `ipc4_get_drv` and it will no longer contain `struct sof_man_fw_desc` of a loadable module.